### PR TITLE
Feat/neon files/sort folders

### PR DESCRIPTION
**Pull request by Leptopoda**
- sort folders first (not configurable)
- hide hidden files (configurable)
- cleanup of some widgets

### CODE REVIEW

File: packages/neon/neon_files/lib/widgets/file_list_tile.dart [hunk 1]
```diff
@@ -10,66 +12,72 @@ class FileListTile extends StatelessWidget {
...
-              width: 10,
-            ),
-            Text(
-              filesize(details.size, 1),
-              style: DefaultTextStyle.of(context).style.copyWith(
-                    color: Colors.grey,
-                  ),
-            ),
+  @override
+  // When the ETag is null it means we are uploading this file right now
```
Comment (provokateurin) [inline packages/neon/neon_files/lib/widgets/file_list_tile.dart:45]: Comment should be one line below

**Review by provokateurin — CHANGES_REQUESTED** [review]
I would be nice to have a warning that shows up when you create a hidden file or folder but you have hidden files not shown. Otherwise the user might get confused why the file/folder was not created.

**Comment by Leptopoda** [comment]
What about doing the warning as a followup (maybe together with https://github.com/nextcloud/neon/issues/535)?

It would currently require adding  the warning to two separate dialogs (folder creation and rename dialog). And I think we should first unify them and them

**Comment by Leptopoda** [comment]
Ah forgot to push one sec

**Pull request merged**